app/js/index.js

```diff
@@ -13,6 +23,55 @@
 		}
 	});
 
+	function newLocationToProcess(locationToProcess) {
+		var locationHostname = urllib.parse(locationToProcess).hostname;
+
+		if (validSlackSubdomain.test(locationHostname)) {
+			var subdomain = validSlackSubdomain.exec(locationHostname);
+			if (subdomain[1] && subdomain[1].length > 1) {
+				var subdomainFiltered = subdomain[1];
+
+				if (subdomainFiltered !== localStorage.getItem(LOCAL_STORAGE_KEY_CURRENT_DOMAIN)) {
+					localStorage.setItem(LOCAL_STORAGE_KEY_CURRENT_DOMAIN, subdomainFiltered);
+				}
+			}
+		} else if (locationHostname === SLACK_DOMAIN) {
+			localStorage.removeItem(LOCAL_STORAGE_KEY_CURRENT_DOMAIN);
+		}
+	}
+
+	function handleLoadIframe(url) {
+		var bodyElement = document.body;
+
+		// Remove all the body's children nodes
+		while (bodyElement.firstChild) {
+			bodyElement.removeChild(bodyElement.firstChild);
+		}
```

Comment:
The comment here explains what is happening but still gives no insight into *why* we are doing it.